modules/dns/dns_api.rb

```diff
@@ -17,7 +17,7 @@ class Api < ::Sinatra::Base
       log_halt(400, "'create' requires fqdn, value, and type parameters") if fqdn.nil? || value.nil? || type.nil?
 
       begin
-        validate_dns_name!(fqdn)
+        validate_dns_name!(fqdn) if type != 'SRV'
```

Comment:
If it's SRV, we can still validate it if we allow underscores. It'd be good to have this validation both here and below.

Reply:
I meant that you should split the validation. Roughly like this:
```ruby
if type == 'SRV'
  validate_srv_name!(fqdn)
else
  validate_dns_name!(fqdn)
end
```